### PR TITLE
Added camelCase aliases for `any`/`anyLast` aggregation functions

### LINKED ISSUE #71401: Aliases for anyLast_respect_nulls / any_respect_nulls with Camel case
`anyLast_respect_nullsStateIf` looks unpleasant. It will be much prettier `anyLastRespectNullsStateIf`.

Also the error message looks weird:
```sql
select anyLast_respect_nullsStateIf(10);

DB::Exception: Aggregate function anyLast_respect_nulls requires single argument. (NUMBER_OF_ARGUMENTS_DOESNT_MATCH)
```

### PR DESCRIPTION
**Pull request by petern48**
Closes #71401

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added aliases `anyRespectNulls`, `firstValueRespectNulls`, and `anyValueRespectNulls` for aggregation function `any`. Also added aliases `anyLastRespectNulls` and `lastValueRespectNulls` for aggregation function `anyLast`. This allows using more natural camel-case-only syntax rather than mixed camel-case/underscore syntax, for example: `SELECT anyLastRespectNullsStateIf` instead of `anyLast_respect_nullsStateIf`.

### Documentation entry for user-facing changes
- [x] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by den-crane** [comment]
@rschu1ze can you mark "can be tested", please?

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 8af8488e46d7c0dc95f454658a83b4187f1b46db with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71403/8af8488e46d7c0dc95f454658a83b4187f1b46db/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71403/8af8488e46d7c0dc95f454658a83b4187f1b46db/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/AggregateFunctions/AggregateFunctionAnyRespectNulls.cpp [hunk 1]
```diff
@@ -223,9 +223,11 @@ void registerAggregateFunctionsAnyRespectNulls(AggregateFunctionFactory & factor
     factory.registerFunction("any_respect_nulls", {createAggregateFunctionAnyRespectNulls, default_properties_for_respect_nulls});
     factory.registerAlias("any_value_respect_nulls", "any_respect_nulls", AggregateFunctionFactory::Case::Insensitive);
     factory.registerAlias("first_value_respect_nulls", "any_respect_nulls", AggregateFunctionFactory::Case::Insensitive);
+    factory.registerAlias("anyRespectNulls", "any_respect_nulls", AggregateFunctionFactory::Case::Sensitive);
```
Comment (rschu1ze) [inline src/AggregateFunctions/AggregateFunctionAnyRespectNulls.cpp:226]: Needs a SQL test.

**Comment by Algunenano** [comment]
We already have `any`, `any_value, `first_value`, and their counterparts: `any_respect_nulls`, `any_value_respect_nulls`, `first_value_respect_nulls`. Do we really need more aliases?

**Comment by den-crane** [comment]
@Algunenano do you like this `anyLast_respect_nullsStateIf` ? 
It's a mix of snake and camel case. And there is no other way.

File: docs/en/sql-reference/aggregate-functions/reference/anylast.md [hunk 2]
```diff
@@ -17,7 +17,10 @@ anyLast(column) [RESPECT NULLS]
 - `column`: The column name. 
 
 :::note
+
```
[the remark this answers is not in the document]
Comment (Algunenano) [reply docs/en/sql-reference/aggregate-functions/reference/anylast.md:20]: > (btw., 1. raises the question what happens if the data contains only NULLs ...)

Same as if there is no data.

Keep in mind that this `RESPECT NULLS` modifier was added to better support some window functions.

The origin of the problem is that CH types by default are not nullable, so the return type of aggregate functions is not nullable too. This poses the problem of what to do when there is no data, which other databases solve by returning NULL and CH by returning the default value (most of the time).

Obviously there are people that really want it to return NULL in those cases, which is why -OrNull modifier was added. But keep in mind that this is for no-data cases, not when you actually want nulls.

And then window functions came in, and the standard declares the option to use `RESPECT NULLS` / `IGNORE NULLS`. For those cases, where we actually need the aggregate function to process the NULLS (not ignore them or do something special when there is no data) some **different** aggregate functions were added. They look the same, but `any(n)` and `any(n) RESPECT NULLS` are actually totally different aggregate functions.

[the remark this answers is not in the document]
Comment (den-crane) [reply docs/en/sql-reference/aggregate-functions/reference/anylast.md:20]: RESPECT NULLS / IGNORE NULLS is standard SQL syntax (not specific to Clickhouse).

>The origin of the problem is that CH types by default are not nullable

AFAIK it's related to the fact that AGG functions skip nulls.

**Comment by rschu1ze** [comment]
After all, these are only aliases. I'd say we can add them but we should add them for all affected functions consistently (and there is just a handful of them).

File: src/AggregateFunctions/AggregateFunctionAnyRespectNulls.cpp (hunk 1 above)
Comment (petern48) [reply src/AggregateFunctions/AggregateFunctionAnyRespectNulls.cpp:226]: Will do

**Comment by petern48** [comment]
@rschu1ze I added the tests, doc clarifications, and aliases for the other two functions. I did a grep and found nothing else that used the RESPECT NULLS modifier, so it should just be these four. Let me know if it looks good or not.

**Comment by rschu1ze** [comment]
Thanks. Made some consistency fixups, everything looks fine now.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**